cpp/tests/utilities_tests/multibuffer_memset_tests.cpp

```diff
@@ -0,0 +1,174 @@
+/*
+ * Copyright (c) 2022-2024, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cudf_test/base_fixture.hpp>
+#include <cudf_test/cudf_gtest.hpp>
+#include <cudf_test/type_lists.hpp>
+
+#include <cudf/io/parquet.hpp>
+
+#include <rmm/device_uvector.hpp>
+
+#include <thrust/iterator/counting_iterator.h>
+#include <thrust/iterator/transform_iterator.h>
+
+#include <src/io/utilities/multibuffer_memset.hpp>
+
+#include <type_traits>
+
+template <typename T>
+struct MultiBufferTestIntegral : public cudf::test::BaseFixture {};
+
+TEST(MultiBufferTestIntegral, BasicTest1)
+{
+  std::vector<long> BUF_SIZES{5000};
+  long NUM_BUFS = BUF_SIZES.size();
+
+  // Device init
+  auto stream = cudf::get_default_stream();
+  auto _mr    = rmm::mr::get_current_device_resource();
+
+  // Init Host buffers to 0xEE and 0xFF as padding around it
+  std::vector<uint64_t*> host_bufs(NUM_BUFS, nullptr);
+  for (int i = 0; i < NUM_BUFS; i++) {
+    CUDF_CUDA_TRY(cudaMalloc(&host_bufs[i], (BUF_SIZES[i] + 2000) * 8));
+    CUDF_CUDA_TRY(cudaMemsetAsync(host_bufs[i], 0xFF, (BUF_SIZES[i] + 2000) * 8, stream.value()));
+    CUDF_CUDA_TRY(cudaMemsetAsync(host_bufs[i] + 1000, 0xEE, BUF_SIZES[i] * 8, stream.value()));
```

Comment:
Would recommend using rmm::device_uvector<> or rmm::device_ubuffer as needed instead of raw cuda allocations.

Reply:
Not recommend, require.